src/test/test_psbt.py

```diff
@@ -308,6 +309,46 @@ def test_invalid_args(self):
                      (psbt, 0x0, None)]: # NULL dest
             self.assertEqual(WALLY_EINVAL, wally_psbt_clone_alloc(*args))
 
+        # Populate PSBT with one input and output to test various invalid args for taproot keypaths
+        self.assertEqual(WALLY_OK, wally_psbt_init_alloc(2, 1, 1, 0, 0, psbt))
+        tx_in = pointer(wally_tx_input())
+        self.assertEqual(WALLY_OK, wally_psbt_add_tx_input_at(psbt, 0, 0, tx_in))
+
+        tx_output = pointer(wally_tx_output())
+        ret = wally_tx_output_init_alloc(1234, b'\x59\x59', 2, tx_output)
+        self.assertEqual(WALLY_OK, ret)
+        ret = wally_psbt_add_tx_output_at(psbt, 0, 0, tx_output)
+        self.assertEqual(WALLY_OK, ret)
+
+        pubkey, pubkey_len = make_cbuffer('339ce7e165e67d93adb3fef88a6d4beed33f01fa876f05a225242b82a631abc0')
+        merkle, merkle_len = make_cbuffer('00' * 32)
+        fpr, fpr_len = make_cbuffer('00' * 4)
+        child_path = ctypes.c_uint32(0)
+        child_path = ctypes.pointer(child_path)
+        child_path_len = 1
+        index = 0
+        flags = 0
+
+        valid_arg = (psbt, index, flags, pubkey, pubkey_len, merkle, merkle_len, fpr, fpr_len, child_path, child_path_len)
+        invalid_args = [
+            (None, index, flags, pubkey, pubkey_len, merkle, merkle_len, fpr, fpr_len, child_path, child_path_len), # NULL psbt
+            (psbt, 1, flags, pubkey, pubkey_len, merkle, merkle_len, fpr, fpr_len, child_path, child_path_len), # invalid index
```

Comment:
The fact that this test fails while the index check is incorrect suggests something else is wrong in the args. If you set index to 0 without changing anything else, this test should fail. Does that happen? if not, you need to correct whichever other argument is causing it to fail throughout the test cases.

Reply:
I think it's due to out of bounds access in wally_psbt_add_input_taproot_keypath triggering a further WALLY_EINVAL